### PR TITLE
Use `inline` to define constant in header

### PR DESCRIPTION
**Pull request by ChrisThrasher**

### CODE REVIEW

**Review by eXpl0it3r — APPROVED** [review]
I think that's correct, but didn't fully understand the reasoning and none was given otherwise.

**Comment by ChrisThrasher** [comment]
The use of `inline` here is a C++17 feature that lets us avoid defining these constants in a particular translation unit.

File: include/SFML/Audio/SoundFileFactory.hpp [hunk 1]
```diff
@@ -152,8 +152,8 @@ class SFML_AUDIO_API SoundFileFactory
     // Static member data
     ////////////////////////////////////////////////////////////
     // NOLINTBEGIN(readability-identifier-naming)
-    static ReaderFactoryArray s_readers; //!< List of all registered readers
-    static WriterFactoryArray s_writers; //!< List of all registered writers
+    static inline ReaderFactoryArray s_readers; //!< List of all registered readers
```
Comment (kimci86) [inline include/SFML/Audio/SoundFileFactory.hpp:155]: I don't see the point. I would rather see this removed from header and put in anonymous namespace in cpp file.

Comment (ChrisThrasher) [reply include/SFML/Audio/SoundFileFactory.hpp:155]: These constants are also used in SoundFileFactory.inl so moving them to the anonymous namespace of SoundFileFactory.cpp is not a possibility.

**Pull request merged**